packages/graph/hash_graph/lib/graph/src/types/schema/object.rs

```diff
@@ -59,17 +63,24 @@ impl<V, const MIN: usize> Object<V, MIN> {
 
     fn validate(&self) -> Result<(), ValidationError> {
         let num_properties = self.properties().len();
-        if num_properties < MIN {
-            return Err(ValidationError::MismatchedPropertyCount {
+        ensure!(
+            num_properties >= MIN,
+            ValidationError::MismatchedPropertyCount {
                 actual: num_properties,
                 expected: MIN,
-            });
-        }
-        for uri in self.required() {
-            if !self.properties().contains_key(uri) {
-                return Err(ValidationError::MissingRequiredProperty(uri.clone()));
             }
+        );
+
+        for uri in self.required() {
+            ensure!(
+                self.properties().contains_key(uri),
+                ValidationError::MissingRequiredProperty(uri.clone())
+            );
+        }
+        for (base_uri, referenced) in self.properties() {
```

Comment:
This is a `reference` rather than a `referenced` right?

Reply:
(As in the object _is_ the reference, not the thing being referenced)

Reply:
5970aad2fe762a8cf33b64608ca473adeb46e75a